File: Legacy-Folder/Backend/HRMSWebApi/HRMS.Domain/Entities/HRClearance.cs
namespace HRMS.Domain.Entities
{
    public class HRClearance : BaseEntity
    {

        public int ResignationId { get; set; }
        public decimal AdvanceBonusRecoveryAmount { get; set; }
        public string ServiceAgreementDetails { get; set; }
        public decimal CurrentEL { get; set; }
        public int NumberOfBuyOutDays { get; set; }
        public bool ExitInterviewStatus { get; set; }
        public string ExitInterviewDetails { get; set; }
        public string Attachment { get; set; }
        public string FileOriginalName { get; set; }

    }
}

File: hrms-backend/app/Models/HRClearance.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class HRClearance extends Model
{
    protected $table = 'hr_clearance';
    protected $primaryKey = 'Id';
    public $timestamps = false;

    protected $fillable = [
        'ResignationId', 'AdvanceBonusRecoveryAmount', 'ServiceAgreementDetails', 
        'CurrentEL', 'NumberOfBuyOutDays', 'ExitInterviewStatus', 'ExitInterviewDetails', 
        'Attachment', 'FileOriginalName', 'CreatedBy', 'CreatedOn', 'ModifiedBy', 'ModifiedOn'
    ];

    protected $casts = [
        'AdvanceBonusRecoveryAmount' => 'decimal:2',
        'CurrentEL' => 'decimal:2',
        'ExitInterviewStatus' => 'boolean',
        'CreatedOn' => 'datetime',
        'ModifiedOn' => 'datetime',
    ];

    // Relationship: HR Clearance belongs to Resignation
    public function resignation()
    {
        return $this->belongsTo(Resignation::class, 'ResignationId', 'Id');
    }
}
